File: WebAPI/WebAPI/Models/Category.cs
namespace WebAPI.Models;

public class Category
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public List<SubCategory> SubCategories { get; set; }
    public AppUser Author { get; set; }
    public Guid AuthorId { get; set; }
    public List<Article> Articles { get; set; }
}

File: client/src/Pages/Articles/Sidebar/Category/Category.js
import React, { useState } from 'react'
import ArtilceLink from '../ArticleLink/ArtilceLink'
import SubCategory from '../SubCategory/SubCategory'

export default function Category({ category }) {
    const [isOpen, setIsOpen] = useState(false)

    if(!isOpen){
        return <li onClick={()=>setIsOpen(true)}>{category.name}</li>
    }
    return (
        <>
            <li onClick={()=>setIsOpen(false)}>
                {category.name}
            </li>
            <ul>
                {category.subcategories.map(sub=>(<SubCategory key={sub.id} subcategory={sub}/>))}
                <ArtilceLink articles={category.articles}/>
            </ul>
        </>
    )
}
